### PR TITLE
cleanup local job_id directory upon clean exit

### PR DESCRIPTION
**Pull request by bmc-msft**

### CODE REVIEW

File: src/agent/onefuzz-agent/src/local/libfuzzer_fuzz.rs [hunk 1]
```diff
@@ -51,7 +51,9 @@ pub fn build_fuzz_config(args: &clap::ArgMatches<'_>, common: CommonConfig) -> R
 
 pub async fn run(args: &clap::ArgMatches<'_>) -> Result<()> {
     let common = build_common_config(args, true)?;
+    register_cleanup(common.job_id)?;
     let config = build_fuzz_config(args, common)?;
+
```
Comment (bmc-msft) [inline src/agent/onefuzz-agent/src/local/libfuzzer_fuzz.rs:56]:
```suggestion
```

File: src/agent/onefuzz-agent/src/local/radamsa.rs [hunk 2]
```diff
@@ -18,7 +18,9 @@ use uuid::Uuid;
 
 pub async fn run(args: &clap::ArgMatches<'_>) -> Result<()> {
     let common = build_common_config(args, true)?;
+    register_cleanup(common.job_id)?;
     let fuzz_config = build_fuzz_config(args, common.clone())?;
+
```
Comment (bmc-msft) [inline src/agent/onefuzz-agent/src/local/radamsa.rs:23]:
```suggestion
```

File: src/agent/onefuzz-agent/src/local/generic_analysis.rs [hunk 3]
```diff
@@ -61,6 +62,7 @@ pub fn build_analysis_config(
 
 pub async fn run(args: &clap::ArgMatches<'_>) -> Result<()> {
     let common = build_common_config(args, true)?;
+    register_cleanup(common.job_id)?;
```
Comment (chkeita) [inline src/agent/onefuzz-agent/src/local/generic_analysis.rs:65]: can we make the behavior configurable with a paramter. I used 'keep_job_dir' in my local run ui PR

**Review by chkeita — APPROVED** [review]

File: src/agent/onefuzz-agent/src/local/common.rs [hunk 4]
```diff
@@ -162,6 +167,30 @@ pub fn get_synced_dirs(
     Ok(dirs?)
 }
 
+fn recursive_remove(path: &Path) -> Result<()> {
```
Comment (ranweiler) [inline src/agent/onefuzz-agent/src/local/common.rs:170]: It seems like this is recreating `remove_dir_all()`, while also calling it. What's the motivation here?

AFAICT, the differences are that we (1) include paths in some of the error messages, when recursively deleting, (2) we allow each recursive remove call to possibly fail one time due to some sort of race.

Comment (bmc-msft) [reply src/agent/onefuzz-agent/src/local/common.rs]: The issue I was experiencing was due to Windows recursion.  I've updated the impl to just use the [remove_dir_all](https://docs.rs/remove_dir_all/0.7.0/remove_dir_all/) crate which handles the issues in a more coherent fashion.

**Review by ranweiler — APPROVED** [review]

**Pull request merged**